### PR TITLE
Fix windows delta updates

### PR DESCRIPTION
**Pull request by mherrmann**
Resolves https://github.com/brave/brave-browser/issues/18696

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by goodov — APPROVED** [review]

**Review by goodov — COMMENTED** [review]
doesn't work for yet

**Review by mihaiplesa — APPROVED** [review]

File: BUILD.gn [hunk 1]
```diff
@@ -57,24 +57,24 @@ if (!is_ios) {
...
 if (is_win) {
   # create_installer_archive.py assumes that last_chrome_installer folder has
-  # courgette.exe.
+  # courgette.exe. The original courgette.exe runs out of memory on Brave's
+  # binaries. As of January 2022, this seems to be a known issue and was
+  # confirmed by Chromium developers. The recommended workaround is to use
+  # another tool, Zucchini, as a drop-in replacement.
   if (build_delta_installer) {
-    copy("copy_courgette_binary") {
+    copy("copy_zucchini_binary") {
```
Comment (goodov) [inline BUILD.gn:65]: how about `copy_zucchini_binary_as_courgette`?

File: BUILD.gn [hunk 2]
```diff
@@ -57,24 +57,24 @@ if (!is_ios) {
...
       visibility = [ ":mini_installer_dependencies" ]
-      sources = [ "$root_out_dir/courgette.exe" ]
-      outputs = [ "$root_out_dir/$last_chrome_installer/{{source_file_part}}" ]
-      deps = [
-        "//courgette",
-        "//courgette:copy_courgette_binaries",
-      ]
+      sources = [ "$root_out_dir/zucchini.exe" ]
+      outputs = [ "$root_out_dir/$last_chrome_installer/courgette.exe" ]
+      deps = [ "//components/zucchini:zucchini" ]
```
Comment (goodov) [inline BUILD.gn:69]: just `//components/zucchini`

**Review by goodov — APPROVED** [review]

**Comment by mherrmann** [comment]
The delta update file `brave_installer-delta-x64.exe` produced by the latest CI build can successfully delta-update in my local dev environment. The next step is to upload it to a staging server and perform an update check from there.

**Pull request merged**